From 3fc30e4956d48bec58c62129f165ac5bff6627f4 Mon Sep 17 00:00:00 2001
From: Neil Armstrong <narmstrong@baylibre.com>
Date: Thu, 23 Jul 2020 14:18:22 +0200
Subject: [PATCH] fixup! drm/panfrost: add Amlogic GPU integration quirks

---
 drivers/gpu/drm/panfrost/panfrost_drv.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/panfrost/panfrost_drv.c b/drivers/gpu/drm/panfrost/panfrost_drv.c
index 162354990bcb9..b0104322c242a 100644
--- a/drivers/gpu/drm/panfrost/panfrost_drv.c
+++ b/drivers/gpu/drm/panfrost/panfrost_drv.c
@@ -670,8 +670,6 @@ static const struct panfrost_compatible default_data = {
 static const struct panfrost_compatible amlogic_gxm_data = {
 	.num_supplies = ARRAY_SIZE(default_supplies),
 	.supply_names = default_supplies,
-	.num_pm_domains = 1, /* optional */
-	.pm_domain_names = NULL,
 	.requires_external_reset = true,
 	.vendor_quirks = panfrost_gpu_amlogic_quirks,
 };
@@ -679,8 +677,6 @@ static const struct panfrost_compatible amlogic_gxm_data = {
 static const struct panfrost_compatible amlogic_g12a_data = {
 	.num_supplies = ARRAY_SIZE(default_supplies),
 	.supply_names = default_supplies,
-	.num_pm_domains = 1, /* optional */
-	.pm_domain_names = NULL,
 	.vendor_quirks = panfrost_gpu_amlogic_quirks,
 };
 
